explainaboard/loaders/file_loader.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+import csv
+from ctypes import Union
+from dataclasses import dataclass
+from io import StringIO
+import json
+from typing import Any, Callable, Iterable, List, Optional
+
+from enum import Enum
+
+from explainaboard.constants import Source
+
+
+class FileLoader:
+    def __init__(
+        self,
+        fields: List[FileLoaderField] = None,
+        use_idx_as_id: bool = True,
+        id_field_name: Optional[str] = None,
+    ) -> None:
+        self._fields = fields or []
+        self._use_idx_as_id = use_idx_as_id
+        self._id_field_name = id_field_name
+
+        # validations
+        if self._use_idx_as_id and self._id_field_name:
+            raise ValueError("id_field_name must be None when use_idx_as_id is True")
+        src_names = [field.src_name for field in self._fields]
+        target_names = [field.target_name for field in self._fields]
+        if len(src_names) != len(set(src_names)):
+            raise ValueError("src_name must be unique")
+        if len(target_names) != len(set(target_names)):
+            raise ValueError("target_name must be unique")
+
+    @staticmethod
+    def parse_data(data: Any, field: FileLoaderField) -> Any:
+        if field.parser:
+            return field.parser(data)
+        if field.strip_before_parsing:
```

Comment:
Shouldn't this before the previous if statement, esp given the naming?

Reply:
`parser` is intended for custom parsers. If a custom parser is provided, everything else will have no effect (dtype type conversion and `strip_before_parsing`). I think this behavior may be more intuitive? because if `strip()` is called on something other than str, the user will get an error message and it may not be obvious what causes that error.

Reply:
OK, sounds good.